src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -161,18 +161,41 @@ def _configure(self):
 
             self._lrms_cores_per_node -= self._lrms_gpus_per_node
 
-            # since we just changed this fundamental setting, we need to
-            # recreate the nodelist.
-            self.nodes = []
-            for node, node_uid in self._lrms_node_list:
-                self.nodes.append({
-                    'name' : node,
-                    'uid'  : node_uid,
-                    'cores': [rpc.FREE] * self._lrms_cores_per_node,
-                    'gpus' : [rpc.FREE] * self._lrms_gpus_per_node,
-                    'lfs'  :              self._lrms_lfs_per_node,
-                    'mem'  :              self._lrms_mem_per_node
-                })
+        # since we just changed this fundamental setting, we need to
+        # recreate the nodelist.
+        self.nodes = []
+        for node, node_uid in self._lrms_node_list:
+
+            node_entry = {'name'   : node,
+                          'uid'    : node_uid,
+                          'cores'  : [rpc.FREE] * self._lrms_cores_per_node,
+                          'gpus'   : [rpc.FREE] * self._lrms_gpus_per_node,
+                          'lfs'    :              self._lrms_lfs_per_node,
+                          'mem'    :              self._lrms_mem_per_node}
+
+            # summit
+            if self._lrms_cores_per_node > 40:
+
+                # Summit cannot address the last core of the second socket at
+                # the moment, so we mark it as `DOWN` and the scheduler skips
+                # it.  We need to check the SMT setting to make sure the right
+                # logical cores are marked.  The error we see on those cores is:
+                # "ERF error: 1+ cpus are not available"
+                #
+                # This is related to the known issue listed on
+                # https://www.olcf.ornl.gov/for-users/system-user-guides \
+                #                          /summit/summit-user-guide/
+                #
+                # "jsrun explicit resource file (ERF) allocates incorrect
+                # resources"
+                #
+                smt = self._lrms_info.get('smt', 1)
+                for s in [1]:
+                    for i in range(smt):
+                        idx = s * 21 * smt + i
+                        node_entry['cores'][idx] = rpc.DOWN
```

Comment:
This makes sure that the cores used by the system are not considered for placing, right?

Reply:
correct - but it is not system use we are hiding, but a bug in slurm which invalidates addresses pointing to that core (see comment above the loop).